lib/active_merchant/billing/gateways/datatrans.rb

```diff
@@ -26,32 +36,32 @@ def purchase(money, payment, options = {})
       end
 
       def authorize(money, payment, options = {})
-        post = add_payment_method(payment)
-        post[:refno] = options[:order_id].to_s if options[:order_id]
+        post = { refno: options.fetch(:order_id, '') }
+        add_payment_method(post, payment)
         add_currency_amount(post, money, options)
         add_billing_address(post, options)
         post[:autoSettle] = options[:auto_settle] if options[:auto_settle]
         commit('authorize', post)
       end
 
       def capture(money, authorization, options = {})
-        post = { refno: options[:order_id]&.to_s }
-        transaction_id, = authorization.split('|')
+        post = { refno: options.fetch(:order_id, '') }
+        transaction_id = authorization.split('|').first
         add_currency_amount(post, money, options)
-        commit('settle', post, { transaction_id: transaction_id, authorization: authorization })
+        commit('settle', post, { transaction_id: transaction_id })
       end
 
       def refund(money, authorization, options = {})
-        post = { refno: options[:order_id]&.to_s }
-        transaction_id, = authorization.split('|')
+        post = { refno: options.fetch(:order_id, '') }
+        transaction_id = authorization.split('|').first
         add_currency_amount(post, money, options)
         commit('credit', post, { transaction_id: transaction_id })
       end
 
       def void(authorization, options = {})
         post = {}
-        transaction_id, = authorization.split('|')
-        commit('cancel', post, { transaction_id: transaction_id, authorization: authorization })
+        transaction_id = authorization.split('|').first
+        commit('cancel', post, { transaction_id: transaction_id })
```

Comment:
💬 
Just wondering why authorization was be deleted from the commit method arguments.
Are why sure it should be removed?

Reply:
Previous PR @Heavyblade clarify me, that I should not pass it.

The idea was: If Datatrans is not providing me an authorization value (or any related parameter), I want to have a value so, I will pass it from the previous authorization to the void and use it as a reference. 

The thing is we must not modify the gateway behavior if the gateway doesn't provide it we must not. This should be a reference to that specific transaction, (a void transaction), if the gateway is not returning one it's something the merchant must deal with.

We pass the transaction_id (a part of the authorization value) because it's required in the capture, and refund and void transactions.